### DISCUSSION TITLE
Desktop version error: CoIncrementMTAUsage could not be located

### DISCUSSION THREAD

**Discussion by GeorgeL59** [ruffle-rs/ruffle#16095] [Q&A] [answered]
I just downloaded the desktop version and tried to run it, but I get the following error message:

ruffle.exe - Entry Point Not Found
---------------------------
The procedure entry point CoIncrementMTAUsage could not be located in the dynamic link library ole32.dll. 
---------------------------
OK   

Environment: 2008-as Dell Latitude D830, Windows 7 SP1 64-bit

I do have two ole32.dll in the Windows directory:
\System32     v6.1.7601.17514      2038 KB
\SysWOW64   v6.1.7601.17514     1381 KB

Any idea, how could I make Ruffle work? I just desperately want to run a SWF game that I used to play often when flash players were standard part of the computing world.

**Comment by torokati44** [comment, marked as answer]
For an explanation of why this happens, see: https://github.com/ruffle-rs/ruffle/issues/15779
You could try using the original standalone Flash Player instead, such as this one: https://archive.org/details/flashplayer32_0r0_363_win_sa

**Reply by GeorgeL59** [reply]
Wow, that was fast, and it works! Unfortunately I just found out, that the swf file that I saved on my pc long time ago was only just the 1st phase of the online game, that probably triggered another one once it finished playing and so forth. 

Do you have any idea how can you make a swf based flash player game working through the Internet Archive? Or that platform was just never designed for acting as a full blown web server?

Here is the link that I am talking about:
https://web.archive.org/web/20160404035349/https://deti.mil.by/templates/swf/Pistol/index.swf

**Reply by torokati44** [reply]
After installing the Ruffle browser extension, the content on your link seems to work fine for me.
At least in Chrome.
In Firefox, there's an issue where sometimes the Wayback Machine's own (slightly outdated) Ruffle takes over the browser extension, and it doesn't work with that version yet.

**Reply by GeorgeL59** [reply]
I don't have Chrome on my laptop, you're right, I do have Firefox and that's the one that I have problem with, only the cartridge of the pistol comes up, but I am unable to start the game by clicking on the Start dot. I do have Opera, the latest version, when I am trying to install the Ruffle, I get the message, that I have to update to the latest version of Opera, I also have Brave, both of them Chrome based if I am not msitaken, I am getting the same message, that I have update Brave, when in fact it is the latest version. So, the only browser, that I managed to install Ruffle on was Firefox, but that is not working with the Wayback Archive server unfortunately.

By any chance, do you know if there is any way to save all swf files of the game locally on your hard drive, so that you could play the game with the locally installed Adobe Flash player? As I wrote earlier, I managed to download the starting swf (index.swf), but that's it.

**Reply by torokati44** [reply]
WayBack Machine can show you all the files it archived under that site: [https://web.archive.org/web/*/https://deti.mil.by/templates/swf/*](https://web.archive.org/web/*/https://deti.mil.by/templates/swf/*).

**Reply by danielhjacobs** [reply]
I'm curious, when you say it's the "latest version", what version is it actually? The latest version of Opera for Windows 7 is version 95, based on Chrome 109. The latest version of Brave for Windows 7 is Release v1.47.186, also based on Chrome 109.

**Reply by GeorgeL59** [reply]
"I'm curious, when you say it's the "latest version", what version is it actually"

I checked it again in Opera (I already uninstalled Brave), I went to the website of Ruffle and clicked on the left button, the one for Chrome, and in the next page I didn't get the error message I got yesterday, but the install button on the right was greyed out, so it was not clickable. I also checked the version of my Opera, it is v95.

**Comment by GeorgeL59** [comment]
> WayBack Machine can show you all the files it archived under that site: [https://web.archive.org/web/_/https://deti.mil.by/templates/swf/_](https://web.archive.org/web/*/https://deti.mil.by/templates/swf/*).

Wow! Now Attila, you just made my day with this revelation! :) I managed to download the pistol game and it works like a charm (with zero delay due to the fact that it is run locally), but I also managed to run the game using Firefox on the Wayback archive page by changing the URL path to end with "/MakarovS.swf" instead of the original path /index.swf.    But thanks to you from now on I'll play it locally because it is way more sensitive and responsive than playing it from the Internet archive server.

Attila, thanks again for the cool solution!

**Comment by GeorgeL59** [comment]
> I'm curious, when you say it's the "latest version", what version is it actually? The latest version of Opera for Windows 7 is version 95, based on Chrome 109. The latest version of Brave for Windows 7 is Release v1.47.186, also based on Chrome 109.
